uber/automated_emails.py

```
@@ -1,75 +1,290 @@
+            log.error('error sending {!r} email to {}', self.subject, model_instance.email, exc_info=True)
+            raise
+
+    @property
+    def when_txt(self):
+        """
+        Return a textual description of when the date filters are active for this email category.
+        """
+
+        return '\n'.join([filter.active_when for filter in self.when])
+
+
+class SendAllAutomatedEmailsJob:
+
+    # save information about the last time the daemon ran so that we can display stats on things like
+    # unapproved emails/etc
+    last_result = dict()
+
+    run_lock = threading.Lock()
 
     @classmethod
-    def send_all(cls, raise_errors=False):
-        if not c.AT_THE_CON and (c.DEV_BOX or c.SEND_EMAILS):
-            with Session() as session:
-                approved = {ae.subject for ae in session.query(ApprovedEmail)}
-                all_sent = set(session.query(Email.model, Email.fk_id, Email.ident))
-                for model, lister in cls.queries.items():
-                    for inst in lister(session):
-                        sleep(0.01)  # throttle CPU usage
-                        for rem in cls.instances.values():
-                            if isinstance(inst, rem.model) and (not rem.needs_approval or rem.subject in approved):
-                                if rem.should_send(inst, all_sent):
-                                    rem.send(inst, raise_errors=raise_errors)
+    def send_all_emails(cls, raise_errors=False):
+        """ Helper method to start a run of our automated email processing """
+        cls().run(raise_errors)
+
+    @timed
+    def run(self, raise_errors=False):
+        """
+        Do one run of our automated email service.  Call this periodically to send any emails that should go out
+        automatically.
+
+        This will NOT run if we're on-site, or not configured to send emails.
+
+        :param raise_errors: If True, exceptions are squashed during email sending and we'll try the next email.
```

I think this comment is wrong...? `raise_errors` seems to squash exceptions only when set to False.